site/public/js/forum.js

```diff
@@ -2104,3 +2104,55 @@ $(() => {
 function scrollThreadListTo(element){
   $(element).get(0).scrollIntoView({behavior: "smooth", block: "center"});
 }
+
+//Only used by the posters and only on recent posts (60 minutes since posted)
+function sendAnnouncement(thread_post_content, id){
+  $(".pin-and-email-message").attr('disabled','disabled');
+  $.ajax({
+    type: 'POST',
+    url: buildCourseUrl(['forum', 'make_announcement']),
+    data: {"id": id, 'csrf_token': window.csrfToken},
+    success: function(data){
+      pinAnnouncement(id, 1, window.csrfToken);
+      window.location.reload();
+    },
+  });
```

Comment:
This needs error handling through Ajax but also checks to see if the JsonResponse was Success or Fail.

Reply:
For some reason when I add the check on if JsonResponse was sucess or fail, the success message doesn't show up at the top... But I included the check anyway, because the user does have some kind of indication with how the button disappears and there is no error message.